lib/twitter/rest/media.rb

```diff
@@ -0,0 +1,122 @@
+      #
+      # @see https://dev.twitter.com/rest/reference/post/media/upload.html
+      # @note This is only for small files, use the chunked upload for larger ones.
+      # @rate_limited Yes
+      # @authentication Requires user context
+      # @raise [Twitter::Error::Unauthorized] Error raised when supplied user credentials are not valid.
+      # @return [Integer] The media_id of the uploaded file.
+      # @param file [File] An image file (PNG, JPEG or GIF).
+      # @option options [String] :media_category Category with which to identify media upload. Possible values include tweet_image, tweet_gif, and tweet_video.
+      def upload_media_simple(file, options = {})
+        Twitter::REST::Request.new(self,
+                                   :multipart_post,
+                                   'https://upload.twitter.com/1.1/media/upload.json',
+                                   key: :media,
+                                   file: file,
+                                   **options).perform[:media_id]
+      end
+
+      # Upload a media file to twitter in chunks
+      #
+      # @see https://dev.twitter.com/rest/reference/post/media/upload.html
+      # @rate_limited Yes
+      # @authentication Requires user context
+      # @raise [Twitter::Error::Unauthorized] Error raised when supplied user credentials are not valid.
+      # @return [Integer] The media_id of the uploaded file.
+      # @param file [File] An image or video file (PNG, JPEG, GIF, or MP4).
+      # @option options [String] :media_category Category with which to identify media upload. Possible values include tweet_image, tweet_gif, and tweet_video.
+      def upload_media_chunked(file, options = {})
+        media_id = chunked_upload_init(file, options)[:media_id]
+        upload_chunks(media_id, file)
+        poll_status(media_id)
+
+        media_id
+      end
+
+    private
+
+      # Finalize upload and poll status until upload is ready
+      #
+      # @param media_id [Integer] The media_id to check the status of
+      def poll_status(media_id)
+        response = chunked_upload_finalize(media_id)
+        10.times do
+          return unless (info = response[:processing_info])
+          return if info[:state] == 'succeeded'
+
+          raise Twitter::Error::ClientError, 'Upload Failed!' if info[:state] == 'failed'
+
+          sleep info[:check_after_secs]
+
+          response = chunked_upload_status(media_id)
+        end
+
+        raise Twitter::Error::ClientError, 'Max status checks exceeded!'
+      end
+
+      # Initialize a chunked upload
+      #
+      # @param file [File] Media file being uploaded
+      # @param options [Hash] Additional parameters
+      def chunked_upload_init(file, options)
+        Twitter::REST::Request.new(self, :post, 'https://upload.twitter.com/1.1/media/upload.json',
+                                   command: 'INIT',
+                                   media_type: 'video/mp4',
```

Comment:
Without having `media_category: 'tweet_video'` specified it uses the "sync" upload.

This means videos are restricted to a duration between 0:30 and 1:00.
U can read more restrictions up here: https://dev.twitter.com/rest/media/uploading-media#videorecs

Without setting this key I get following error:
```
2017-01-11T11:23:20.793Z 32352 TID-gom2y8e00 WARN: Twitter::Error::BadRequest: Duration too long, maximum:30000, actual:60095 (MediaId: snf:819142603191054338)
```

When adding the `media_category: 'tweet_video'` I was able to upload the video which is slightly longer than 1 minute (60095ms).